src/MongoDB.Driver/UpdateDefinition.cs

```diff
@@ -26,13 +27,25 @@ namespace MongoDB.Driver
     /// <typeparam name="TDocument">The type of the document.</typeparam>
     public abstract class UpdateDefinition<TDocument>
     {
+        /// <summary>
+        /// Renders the update to a <see cref="BsonValue"/>.
+        /// </summary>
+        /// <param name="documentSerializer">The document serializer.</param>
+        /// <param name="serializerRegistry">The serializer registry.</param>
+        /// <returns>A <see cref="BsonValue"/>.</returns>
+        public abstract BsonValue Render(IBsonSerializer<TDocument> documentSerializer, IBsonSerializerRegistry serializerRegistry);
+
         /// <summary>
         /// Renders the update to a <see cref="BsonDocument"/>.
         /// </summary>
         /// <param name="documentSerializer">The document serializer.</param>
         /// <param name="serializerRegistry">The serializer registry.</param>
         /// <returns>A <see cref="BsonDocument"/>.</returns>
-        public abstract BsonDocument Render(IBsonSerializer<TDocument> documentSerializer, IBsonSerializerRegistry serializerRegistry);
+        public BsonDocument RenderAsBsonDocument(IBsonSerializer<TDocument> documentSerializer, IBsonSerializerRegistry serializerRegistry)
```

Comment:
I would remove this method.

Callers can simply cast the result themselves (if they require the value to be a BsonDocument).